core/logic/src/main/java/org/apache/syncope/core/logic/AuditLogic.java

```diff
@@ -41,60 +35,31 @@
 import java.util.stream.Collectors;
 
 @Component
-public class AuditLogic extends AbstractTransactionalLogic<AuditTO> {
-    private static final Logger LOG = LoggerFactory.getLogger(AuditLogic.class);
-
+public class AuditLogic extends AbstractTransactionalLogic<AuditEntryTO> {
     @Autowired
     private AuditDataBinder binder;
 
     @Autowired
     private AuditDAO auditDAO;
 
     @Override
-    protected AuditTO resolveReference(final Method method, final Object... args) throws UnresolvedReferenceException {
-        String who = null;
-
-        if (ArrayUtils.isNotEmpty(args)) {
-            for (int i = 0; who == null && i < args.length; i++) {
-                if (args[i] instanceof String) {
-                    who = (String) args[i];
-                } else if (args[i] instanceof AuditTO) {
-                    who = ((AuditTO) args[i]).getWho();
-                }
-            }
-        }
-
-        if (who != null) {
-            try {
-                return binder.getAuditTO(auditDAO.findByUsername(who));
-            } catch (final Throwable throwable) {
-                LOG.debug("Unresolved reference", throwable);
-                throw new UnresolvedReferenceException(throwable);
-            }
-        }
-
+    protected AuditEntryTO resolveReference(final Method method, final Object... args)
+        throws UnresolvedReferenceException {
         throw new UnresolvedReferenceException();
     }
 
     @PreAuthorize("hasRole('" + StandardEntitlement.AUDIT_SEARCH + "')")
     @Transactional(readOnly = true)
-    public Pair<Integer, List<AuditTO>> search(
-        final SearchCond searchCond,
-        final int page, final int size, final List<OrderByClause> orderBy,
-        final String realm,
-        final boolean details) {
-
-        SearchCond effectiveSearchCond = searchCond == null ? auditDAO.getAllMatchingCond() : searchCond;
-        int count = auditDAO.count(RealmUtils.getEffective(
-            AuthContextUtils.getAuthorizations().get(StandardEntitlement.AUDIT_SEARCH), realm), effectiveSearchCond);
+    public Pair<Integer, List<AuditEntryTO>> search(
+        final String key,
+        final int page,
+        final int size,
+        final List<OrderByClause> orderByClauses) {
 
-        List<AuditEntry> matching = auditDAO.search(RealmUtils.getEffective(
-            AuthContextUtils.getAuthorizations().get(StandardEntitlement.AUDIT_SEARCH), realm),
-            effectiveSearchCond, page, size, orderBy);
-        List<AuditTO> result = matching.stream().
-            map(binder::getAuditTO).
+        List<AuditEntry> matching = auditDAO.findByEntityKey(key, page, size, orderByClauses);
+        List<AuditEntryTO> results = matching.stream().
+            map(audit -> binder.returnAuditTO(binder.getAuditTO(audit))).
             collect(Collectors.toList());
-
-        return Pair.of(count, result);
+        return Pair.of(results.size(), results);
```

Comment:
why `results.size()` and not (the dismissed) `count()`?

Reply:
You're right! I missed the fact count() does not take into account sizing and paging flags and so `count()` might in fact return a different result than `result.size()`. I'll correct this. Thanks!